plugins/module_utils/foreman_helper.py

```diff
@@ -26,6 +26,34 @@
     APYPIE_IMP_ERR = traceback.format_exc()
 
 
+def _entity_spec_helper(spec):
+    """Extend an entity spec by adding entries for all alt_ and flat_names.
+    Extract ansible compatible argument_spec on the way.
+    """
+    entity_spec = {'id': {}}
+    argument_spec = {}
+    for key, value in spec.items():
+        if 'alt_names' in value:
+            value = value.copy()
+            alt_names = value.pop('alt_names')
+            for alt_name in alt_names:
+                entity_spec[alt_name] = value
+        entity_spec[key] = value
+        if 'flat_name' in value:
```

Comment:
I know this `flat` comes from the original `_flatten_value` I wrote, but I think this rather should be `foreman_name` or something?

Reply:
Well i like it to be called 'flat', because it distinguishes between the inflated version of the entity (with dictionaries for related entities) as provided by the user _and_ api responses, and the 'flat' version with only ids as required in the api requests.

Reply:
naming is hard, I can live with either :)